### PR TITLE
Add extra_rustc_flag and extra_exec_rustc_flag

### PR DESCRIPTION
**Pull request by mypmc**
Retake of #1407.
Each new flags are singular, and appended after the flags passed to plural ones.

Use case:

```
build --@rules_rust//:extra_rustc_flags="-C...,-C...,-Z..."

build:dev --@rules_rust//:extra_rustc_flag="-C..."
build:dev --@rules_rust//:extra_rustc_flag="-C..."

build:test --config=dev
build:test --@rules_rust//:extra_rustc_flag="-C..."

build:fuzz --config=test
build:fuzz --@rules_rust//:extra_rustc_flag="-C..."
build:fuzz --@rules_rust//:extra_rustc_flag="-C..."
build:fuzz --@rules_rust//:extra_rustc_flag="-Z..."
```

### CODE REVIEW

File: rust/private/rustc.bzl [hunk 1]
```diff
@@ -834,9 +834,15 @@ def construct_arguments(
     if hasattr(ctx.attr, "_extra_rustc_flags") and not is_exec_configuration(ctx):
         rustc_flags.add_all(ctx.attr._extra_rustc_flags[ExtraRustcFlagsInfo].extra_rustc_flags)
 
+    if hasattr(ctx.attr, "_extra_rustc_flag") and not is_exec_configuration(ctx):
+        rustc_flags.add_all(ctx.attr._extra_rustc_flag[ExtraRustcFlagsInfo].extra_rustc_flags)
+
     if hasattr(ctx.attr, "_extra_exec_rustc_flags") and is_exec_configuration(ctx):
         rustc_flags.add_all(ctx.attr._extra_exec_rustc_flags[ExtraExecRustcFlagsInfo].extra_exec_rustc_flags)
 
+    if hasattr(ctx.attr, "_extra_exec_rustc_flag") and not is_exec_configuration(ctx):
```
Comment (krasimirgg) [inline rust/private/rustc.bzl:843]: shouldn't this check be `if hasattr(ctx.attr, "_extra_exec_rustc_flag") and is_exec_configuration(ctx)`?

File: test/unit/extra_rustc_flags/extra_rustc_flags_test.bzl [hunk 2]
```diff
@@ -58,6 +58,20 @@ extra_rustc_flags_present_test = analysistest.make(
     },
 )
 
+extra_rustc_flag_present_test = analysistest.make(
```
Comment (krasimirgg) [inline test/unit/extra_rustc_flags/extra_rustc_flags_test.bzl:61]: could you also add a similar test for `extra_exec_rustc_flag`

**Comment by krasimirgg** [comment]
How are we going to deal with the Min Bazel version failing CIs again?

**Comment by mypmc** [comment]
> How are we going to deal with the Min Bazel version failing CIs again?

PR for it. https://github.com/bazelbuild/rules_rust/pull/1414

File: rust/private/rustc.bzl (hunk 1 above)
Comment (krasimirgg) [reply rust/private/rustc.bzl]: ptal

**Comment by mypmc** [comment]
@krasimirgg ping; what should I do next??

**Review by krasimirgg — APPROVED** [review]
Latest version looks good.

**Review by UebelAndre — APPROVED** [review]

**Pull request merged**